okonf/facts/systemd.py

```diff
@@ -39,3 +39,20 @@ async def enforce(self, host: Executor) -> bool:
         async with host.lock(f"systemctl-start-service-{self.name}"):
             await host.run(f"sudo systemctl start {self.name}")
         return True
+
+
+class ServiceRestarted(Fact):
+    """Restart a Systemd service."""
+
+    name: str
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    async def enquire(self, host: Executor) -> bool:
+        return False
+
+    async def enforce(self, host: Executor) -> bool:
+        async with host.lock(f"systemctl-restart-service-{self.name}"):
```

Comment:
The name of this lock is different from the one used in `ServiceStarted`, so that instruction could in theory be run in at the same time. It would be better to rename the lock `systemctl-service-{self.name}` in both `ServiceStarted` and `ServiceRestarted`.